jhub_apps/hub_client/hub_client.py

```diff
@@ -125,3 +213,24 @@ def get_services(self):
         r = requests.get(API_URL + "/services", headers=self._headers())
         r.raise_for_status()
         return r.json()
+
+    def get_groups(self):
+        """Returns all the groups in JupyterHub"""
+        r = requests.get(API_URL + "/groups", headers=self._headers())
+        r.raise_for_status()
+        return r.json()
+
+
+def get_users_and_group_allowed_to_share_with(scopes):
+    """Returns a list of users and groups"""
+    hclient = HubClient()
+    users = hclient.get_users()
+    user_names = [user["name"] for user in users]
+    groups = hclient.get_groups()
+    group_names = [group['name'] for group in groups]
+    # TODO: Filter users and groups based on what the user has access to share with
+    # parsed_scopes = parse_scopes(scopes)
+    return {
+        "users": user_names,
```

Comment:
Does it makes sense to always filter the current user out as sharing with yourself does not have much utility?